backend/sftp/sftp.go

```diff
@@ -740,6 +752,10 @@ func NewFs(ctx context.Context, name, root string, m configmap.Mapper) (fs.Fs, e
 		ClientVersion:   "SSH-2.0-" + f.ci.UserAgent,
 	}
 
+	if opt.HostKeyAlgorithms.String() != "" {
```

Comment:
```go
if len(opt.HostKeyAlgorithms) != 0 {
```

would be idiomatic here

Reply:
Appreciate the feedback. Was completely forgetting I could just use standard list operations here but you are correct that is much better!